### PR TITLE
Added match and dump methods to PcapPyBpfProgram

### PR DESCRIPTION
**Pull request by remleduff**
This allows invoking a filter against a packet using pcap_offline_filter

### CODE REVIEW

File: src/pcappy/__init__.py [hunk 1]
```diff
@@ -276,6 +276,20 @@ def netmask(self):
...
+    def match(self, packet, length=None, caplen=None):
+        pkt = str(packet)
+        hdr = pcap_pkthdr()
+        if length is None:
+            length = len(pkt)
+        if caplen is None:
+            caplen = length
+        hdr.len = length
+        hdr.caplen = caplen
+        return pcap_offline_filter(pointer(self._bpf), pointer(hdr), pkt)
```
Comment (remleduff) [inline src/pcappy/__init__.py]: This should probably return:
pcap_offline_filter(pointer(self._bpf), pointer(hdr), pkt) != 0

Comment (allfro) [inline src/pcappy/__init__.py]: You might run into a nasty bug here where `str()` will truncate `packet` when encountering a null byte - especially if you are converting a buffer into a string.

File: src/pcappy/functions.py [hunk 2]
```diff
@@ -144,6 +144,9 @@ def _pcap_unsupported(*args, **kwargs):
 load_func('pcap_compile_nopcap', c_int, [ c_int, c_int, bpf_program_ptr, c_char_p, c_int, c_uint32 ])
 
 
+load_func('bpf_dump', argtypes=[ bpf_program_ptr, c_int ])
```
Comment (allfro) [inline src/pcappy/functions.py]: This statement already exists in `functions.py`.

**Comment by allfro** [comment]
Please see latest commit (98833ccf68a779fdfa97038037a74374a457f01d). Those methods were implemented as `dump`, `is_match` and `is_match2` in `PyPcapBpfProgram`. `is_match2` does what you are trying to achieve with the merge request above.

**Pull request closed without merging**